Sapphirization of OpenAlpr-Android

sungwook-moon opened the pull request:
This PR contains Sapphirization of OpenAlpr-Android which was tested to run both on Android devices (Kinndle/Huawei Mate 9) and cloud servers (AWS EC2).

When running on the cloud server, it needs to install original OpenAlpr (not this openalpr-android) with modified native code to work (links will be added with another PR that shows changes in OpenAlpr repo).

Note that there are Sapphire-Cord code added in this PR which can be huge and confusing. Therefore, another PR was created to show the diff without Sapphire-Core. This PR also contains a git diff text file that shows the changes made to Sapphire-core.

Commit:
2018-2-2 initial commit to create a Androd Studio project.

1. Initial change to make the app work.
2. proguard was commented out as it causes an issue.

Commit:
2018-2-5 Print number of plates and other information.

1. Print number of plates found.
2. Add a scroll bar to the result.

Commit:
Initial check-in for Sapphirization.

Currently fails at ret = object.invoke(rpc.getMethod(), rpc.getParams()); in KernelSererImpl.java

TODO
1. Image file needs to be transferred to the server over network.
2. Update recognizeWithCountryRegionNConfig in AlprSapphire to load an image file from server (need to provide path to the image file).

Commit:
Added file uploading over RMI

TODO:
1. This needs to be optimized to send a compressed picture.
2. RMI to process the image fails due to library access failure on the server at "System.loadLibrary("openlapr-native");" in AlprJNIWrapper.java

Commit:
2018-2-15 OpenAlpr server call first success

1. Add JNI code from OpenAlpr (different from Android version repo): https://github.com/openalpr/openalpr/tree/master/src/bindings/java/src/com/openalpr/jni
2. Add test Main program to test basic OpenAlpr commands in Ubuntu 16.
3. RMI to AWS EC2 instance.
4. Upate Sapphire Kernel to show log message on the console.
5. Update AlprJNIWrapper and related code to use non-Android specific Alpr native API.

Commit:
2018-2-16-Demo-Work

This commit makes the call to Server/Android work. Testing confirmed ~4 seconds for processing 50 plates at Server (~20 seconds in the Android device).

Note that this code needs lots of polishing (and todo for merging code).

Commit:
2018-3-1 Added Processing Entity Button with Image Size Reduction.

1. Added process on device or server button.
2. Reduce the image to the predefined maximun size.

Commit:
2018-3-13-WorkBothForDeviceAndLocalComputer

1. Work with device Sapphire kernel server.
2. Work with local computer in the same internal networks.

Windows need .dll file for openalpr to work (not checked in here).

TODO: Update JNI C++ files to have the same API between Android and server.

Commit:
Support processing at device and server in the same process.

1.  Add region information registration to distinguish processing between device and server.
2. Processing time state display in UI.
3. Support Amazon Kindle.

Note it assumes OMS is running on the same network. To run the computation at Cloud, OMS should run on Cloud and the app will connect to OMS in the cloud. To run the processing on the device, OMS should run on the same network as the device (or device should be reachable from OMS).

Commit:
2018-3-20 Update to the latest Sapphire-core code in master branch of DCAP-Sapphire

1. Synced (copied) Sapphire-core code from https://github.com/Huawei-PaaS/DCAP-Sapphire

2. Removed unnecessary Sapphire-core in the different folder (root folder). The real Sapphire-core code resides under Sample folder, hence, there had been two copies previously (root level one was not used).

Commit:
2018-3-22-Copy of latest Sapphire core code and explicit DM pull request.

This is in case we need to work on configurable DM.

Commit:
2018-3-22 Removed compiler generated files and unused dependencies in root folder.

Commit:
Removed explicit DM.

1. Explicit DM is currently not checked in master branch. It was added for working on a new DM but removed since this can confuse the pull request.
2. Removed generated stub files under policy/stubs.

Commit:
SapphireCoreDiff and revert the channge in KernelObject

1. Revert change made to KernelObject as the chage was due to explicitDM copy (which was discarded in this repo).
2. Check in Sapphire core git difference to showt the changes made in SapphireCore compared to the DCAP repo as of 3/22/2018.

ghost commented:
Wow!  38K lines is going to take a while to figure out how best to review.  I'll give it a try next week.

sungwook-moon commented:
Please refer to another PR for the changelist without Sapphire-core. It has much smaller code: https://github.com/sungwook-moon/openalpr-android/pull/2

Commit:
2018-4-3-UI Change

little adjustment in UI to show the result on the same screen as buttons. Previously, the result plate information was hiding the buttons.
